plugins/ComposeAuth/src/androidMain/kotlin/io/github/jan/supabase/compose/auth/Utils.kt

```diff
@@ -17,4 +18,20 @@ fun getSignInRequest(config: GoogleLoginConfig?): BeginSignInRequest {
     }
     return BeginSignInRequest.builder()
         .setGoogleIdTokenRequestOptions(tokenIdRequestOptions.build()).build()
+}
+
+fun SignInCredential.fetchExtraData(config: GoogleLoginConfig):List<LoginConfig.ExtraData>{
+    if(config.extraData.isNullOrEmpty()) return emptyList()
+
+    config.extraData.forEach {
+        when(it){
+            is GoogleLoginConfig.DisplayName -> it.value = this.displayName
+            is GoogleLoginConfig.FamilyName -> it.value = this.familyName
+            is GoogleLoginConfig.GivenName -> it.value = this.givenName
+            is GoogleLoginConfig.ID -> it.value = this.id
+            is GoogleLoginConfig.ProfilePic -> it.value = this.profilePictureUri.toString()
+        }
+    }
+
+    return config.extraData
```

Comment:
Sorry if I said something confusing earlier, you don't actually have to do that manually, supabase automatically fetches the data. The [other login methods](https://github.com/supabase-community/supabase-kt/blob/a2ebf914d25950b9b78900f1019fab18333c39cb/GoTrue/src/commonMain/kotlin/io/github/jan/supabase/gotrue/GoTrue.kt#L175) just have a JsonObject property to add additional data